portal-network.md

```diff
@@ -47,11 +47,19 @@ TODO
 
 ### Chain History: Headers, Blocks, and Receipts
 
-TODO
+In order to validate requested portal network data, a portal client needs to be able to request headers and validate their inclusion in the canonical header chain. Clients can use validated headers to further validate blocks, uncles, transactions, receipts, and state nodes.
 
-### Canonical Indices: Transactions by Hash and Blocks by Hash
+Normal clients will download every block header to construct the canonical chain. This is unreasonable for a "stateless" client. The "double-batched merkle log accumulator" is the mechanism that enables portal clients to achieve this goal, without requiring them to download every canonical header.
 
-TODO
+When a portal client requests a certain header, the response includes two accumulator proofs. As long as the client maintains an up-to-date view of the chain tip (via the gossip network) it can use the proofs to validate a headers inclusion in the canonical chain. Then, the client can use the header to validate other data retrieved from the portal network.
+
+### Canonical Indices: Transactions by Hash and Blocks by Number
+
+To serve the `eth_getTransactionByHash` and `eth_getBlockByNumber` JSON-RPC API endpoints, clients typically build local indexes as they sync the entire chain. The portal network will need to mimic these indices. This can be acheived by generating unique mappings for transactions and blocks, and then pushing these into to portal network.
+
+Since valid transactions can exist outside of the context of a particular block, we need a mechanism to prove that they were included in a certain block. Likewise, valid blocks can exist as uncles, without proof that they were included in the canonical chain at a certain number. So, a mechanism is required to validate that a given block is canonical at a certain level, and a transaction was included in a canonical block.
+
+In the portal network, the canonical block index is built by mapping block numbers to their respective canonical block hash. The canonical transaction index is built by mapping transaction hashes to their respective canonical block hash and transaction index. These key-value pairs are then pushed into the portal network, similar to other data objects.
```

Comment:
This is close, but I think you need to link this back to the "double-batched" accumulator.  Try to take the following list and expand it into something a bit more descriptive.

- The network stores a mapping of `transaction_hash -> (block_number, transaction_index)`
- In order to "prove" `(block_hash, transaction_index) a client will need to:
    - be able to fetch the header indicated by `block_hash` and verify it is part of the canonical chain via a proof against the "double batched accumulator".
    - fetch the block body, verify it against the header, and then verify that the transaction at `transaction_index` matches the expected transaction hash

The above describes the high level functionality necessary to lookup transactions by their hash.  I think the thing you need to figure out is how to cohesively describe this in accessible language.

Reply:
I updated the description, lmk if you think it's along what you're looking for - or if it needs more detail / adjustment.